### PR TITLE
Add unit tests for org.ojalgo.machine.BasicMachine

### PR DESCRIPTION
**Pull request by sudo-jaa**
Hi,
I've analysed your codebase and seen some gaps in the coverage of BasicMachine.
I've written the tests for the functions within BasicMachine with the help of [Diffblue Cover](https://www.diffblue.com/products).
Hopefully, these tests will help you detect regressions caused by future code changes.

### CODE REVIEW

File: test/org/ojalgo/machine/TestBasicMachine.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        TestUtils.assertEquals("0kB/6threads", new BasicMachine(-1L, 6).toString());
+        TestUtils.assertEquals("0kB/1thread", new BasicMachine(0L, 1).toString());
+        TestUtils.assertEquals("2GB/1thread", new BasicMachine(3_000_000_737L, 1).toString());
+        TestUtils.assertEquals("2GB/17threads", new BasicMachine(3_000_000_737L, 17).toString());
+        TestUtils.assertEquals("10737418291776bytes/1thread", new BasicMachine(10_737_418_291_776L, 1).toString());
+        TestUtils.assertEquals("10737418291776bytes/9threads", new BasicMachine(10_737_418_291_776L, 9).toString());
+    }
+
+    @Test
+    public void testhashCode() {
```
Comment (freddytuxworth) [inline test/org/ojalgo/machine/TestBasicMachine.java]: Capitalize `h` in method name

File: test/org/ojalgo/machine/TestBasicMachine.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package org.ojalgo.machine;
+
+import org.junit.jupiter.api.Test;
+import org.ojalgo.TestUtils;
+
+public class TestBasicMachine {
+    @Test
+    public void testEquals_memoryUnequal() {
+
+        final BasicMachine a = new BasicMachine(-4_756_927_793_668_063_236L, 0);
```
Comment (freddytuxworth) [inline test/org/ojalgo/machine/TestBasicMachine.java]: These huge negative memory values are a little counter-intuitive (although not strictly a problem in the class).

**Comment by freddytuxworth** [comment]
I would also say that the underscores in method names are problematic, but then some other test classes in the project use them so I guess ~~nothing is sacred~~ they don't mind it.

**Comment by petersear** [comment]
Remove all blanks lines within the tests?

**Comment by peterschrammel** [comment]
> I would also say that the underscores in method names are problematic, but then some other test classes in the project use them

That would be an indication that underscores should be used for this project.

File: test/org/ojalgo/machine/TestBasicMachine.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package org.ojalgo.machine;
+
+import org.junit.jupiter.api.Test;
+import org.ojalgo.TestUtils;
+
+public class TestBasicMachine {
```
Comment (peterschrammel) [inline test/org/ojalgo/machine/TestBasicMachine.java:27]: Is it possible to get also tests for users of this class such as `Hardware` or `VirtualMachine`?

Comment (sudo-jaa) [reply test/org/ojalgo/machine/TestBasicMachine.java:27]: We had a few tests for these classes, though not as high coverage as BasicMachine, and also with heavy use of reflection. I'll try to re-factor some of these.

**Review by freddytuxworth — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**